docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+The container workload process defined by the OCI bundle's
+configuration file (`config.json`) is created using the Kata
+Containers [agent](#agent). The agent creates a container environment
+to run the workload in, in a similar manner to that done by the
+[`runc`](https://github.com/opencontainers/runc) OCI runtime, using
+Linux cgroups and namespaces.
+
+Note that the workload is isolated from the VM root environment by its
+surrounding container environment. The VM environment the workload
+container runs in is also isolated from the "outer" host environment
+where the container manager runs.
 
 ## Runtime
 
-`containerd-shim-kata-v2` is a [containerd runtime shimv2](https://github.com/containerd/containerd/blob/v1.4.1/runtime/v2/README.md) implementation and is responsible for handling the `runtime v2 shim APIs`, which is similar to [the OCI runtime specification](https://github.com/opencontainers/runtime-spec) but simplifies the architecture by loading the runtime once and making RPC calls to handle the various container lifecycle commands. This refinement is an improvement on the OCI specification which requires the container manager call the runtime binary multiple times, at least once for each lifecycle command.
+The Kata Containers runtime (or shim), is a [shimv2](#shim-v2-architecture) compatible runtime.
 
-`containerd-shim-kata-v2` heavily utilizes the
-[virtcontainers package](../../src/runtime/virtcontainers/), which provides a generic, runtime-specification agnostic, hardware-virtualized containers library.
+The name of the Kata Containers shim binary is
+[`containerd-shim-kata-v2`](../../src/runtime/cmd/containerd-shim-kata-v2).
 
-### Configuration
+The runtime makes heavy use of the [`virtcontainers`
+package](../../src/runtime/virtcontainers), which provides a generic,
+runtime-specification agnostic, hardware-virtualized containers
+library.
+
+The runtime is responsible for starting the [hypervisor](#hypervisor)
+and it's VM, and communicating with the [agent](#agent) using a
+[ttRPC based protocol](#agent-communications-protocol) over a VSOCK
+socket that provides a communications link between the VM and the
+host. 
+
+This protocol allows the runtime to send container management commands
+to the agent. The protocol is also used to carry the standard I/O
+streams (`stdout`, `stderr`, `stdin`) between the containers and
+container managers (such as CRI-O or containerd).
 
-The runtime uses a TOML format configuration file called `configuration.toml`. By default this file is installed in the `/usr/share/defaults/kata-containers` directory and contains various settings such as the paths to the hypervisor, the guest kernel and the mini-OS image.
+## Utility program
 
-The actual configuration file paths can be determined by running:
-```
-$ kata-runtime --show-default-config-paths
-```
-Most users will not need to modify the configuration file.
+The `kata-runtime` binary is a utility program that provides
```

Comment:
One thing that would help is to explain how the kata-runtime is used in the shimv2 case and how it is used in the crio case.  I think that the kata-runtime binary is only for the SHIMv1 case with CRIO.  I think clearly separating out these two cases throughout the document would help a lot.

Reply:
And Eric's comment server as a point of how kata-runtime / runtime / shim can be confusing

Reply:
This doc only applies to Kata 2.x. Again, we could add extra detail, but to give this PR a chance of ever landing, let's defer that one for now.